### ISSUE TITLE
Gulp4 and Imagemin error

### ISSUE THREAD

**Issue by abhilashdk2016** [imagemin/imagemin#279]
Hi,

I am running a Windows 10 OS and trying to use gulp4 with imagemin to minify some png files. But ran into errors:

Here is my gulpfile:
```
var gulp = require('gulp4'),
    concat = require('gulp-concat'),
    uglify = require('gulp-uglify'),
    myth = require('gulp-myth'),
    jshint = require('gulp-jshint'),
    imagemin = require('gulp-imagemin');

var paths = {
    js: 'app/js/*.js',
    css: 'app/css/*.css',
    images: 'app/img/*.{JPG,jpg,png,gif}',
};

//Styles Task
gulp.task('styles', function(){
    return gulp.src(paths.css)
               .pipe(concat('all.css'))
               .pipe(myth())
               .pipe(gulp.dest('dist'));
});

//Scripts Task
gulp.task('scripts', function() {
    return gulp.src(paths.js)
               .pipe(jshint())
               .pipe(jshint.reporter('default'))
               .pipe(concat('all.js'))
               .pipe(uglify())
               .pipe(gulp.dest('dist'));
});

//Images Task
gulp.task('images', function(){
    return gulp.src(paths.images)
               .pipe(imagemin())
               .pipe(gulp.dest('dist/img'));
});

//Watch Task
gulp.task('watch', function() {
    gulp.watch(paths.css,gulp.series('styles'));
    gulp.watch(paths.js, gulp.series('scripts'));
    gulp.watch(paths.images,gulp.series('images'));
});

//Default task
gulp.task('default', 
gulp.parallel('styles','scripts','images','watch'));
```

Here is the error:
```
[10:57:44] Using gulpfile D:\Documents\Working\Gulp\gulp-book\gulpfile.js
[10:57:44] Starting 'default'...
[10:57:44] Starting 'styles'...
[10:57:44] Starting 'scripts'...
[10:57:44] Starting 'images'...
[10:57:44] Starting 'watch'...
[10:57:45] Finished 'styles' after 874 ms
[10:57:45] Finished 'scripts' after 881 ms
[10:57:45] 'images' errored after 913 ms
[10:57:45] Error in plugin "gulp-imagemin"
Message:
    The system cannot find the path specified.

Details:
    errno: ENOENT
    code: ENOENT
    syscall: spawn D:\Documents\Working\Gulp\gulp-book\node_modules\optipng-bin\vendor\optipng.exe
    killed: false
    stdout:
    stderr: The system cannot find the path specified.

    failed: true
    signal: null
    cmd: D:\Documents\Working\Gulp\gulp-book\node_modules\optipng-bin\vendor\optipng.exe -strip all -clobber -fix -o 3 -out C:\Users\abhilk\AppData\Local\Temp\4f2800d8-3bc0-47e1-95a0-2c9e73133c05 C:\Users\abhilk\AppData\Local\Temp\bf26e857-0623-4783-b68a-c76223792cd3
    timedOut: false
    fileName: D:\Documents\Working\Gulp\gulp-book\app\img\gulp.png
    domainEmitter: [object Object]
    domain: [object Object]
    domainThrown: false

[10:57:45] 'default' errored after 954 ms
[10:57:45] The following tasks did not complete: watch
[10:57:45] Did you forget to signal async completion?
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gulp-book@1.0.0 taskrunner: `gulp4`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the gulp-book@1.0.0 taskrunner script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\abhilk\AppData\Roaming\npm-cache\_logs\2018-02-06T05_27_45_442Z-debug.log
```

package.json file:
```{
  "name": "gulp-book",
  "version": "1.0.0",
  "description": "",
  "main": "gulpfile.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "taskrunner": "gulp4"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "gulp": "^3.9.1",
    "gulp-concat": "^2.6.1",
    "gulp-imagemin": "^4.1.0",
    "gulp-jshint": "^2.1.0",
    "gulp-myth": "^1.1.0",
    "gulp-uglify": "^3.0.0",
    "gulp4": "^4.0.2",
    "imagemin-jpegtran": "^5.0.2",
    "imagemin-pngquant": "^5.0.1",
    "jshint": "^2.9.5",
    "pngquant-bin": "^3.1.1"
  },
  "dependencies": {
    "optipng-bin": "^4.0.0"
  }
}
```
Please let me how I can resolve this. Thanks :)

**Comment by WesselKroos** [comment]
I'm using gulp 3 but got the same error with the latest version (4.1.0) of the gulp-imagemin package. 
A quick fix that worked for me was to revert back to version 3.4.0 of the gulp-imagemin package.

**Comment by abhilashdk2016** [comment]
Yes reverting back to version 3.4.0 worked for me.

**Comment by WesselKroos** [comment]
@abhilashdk2016 Today I've upgraded a gulp project and received the same error again. But now I've remove the package and reinstalled it with the latest version. So this bug seems to be fixed from version 5.0.3. (Keep in mind that I'm using node version 10.x current while using 8.x previously)

**Comment by XhmikosR** [comment]
The error shows that optipng was missing on your system. Usually removing the local node_modules folder and doing `npm i` fixes this, otherwise something (like antivirus) is blocking the file.
